### PR TITLE
[SYCL] Fix one more warning coming from SYCL headers

### PR DESCRIPTION
**Pull request by fadeeval**
Fixed "warning: '&&' within '||' [-Wlogical-op-parentheses]" in types.hpp: it wasn't caught by the originally added test, because it is triggered only if corresponding template is instantiated.

Adjusted clang arguments in existing test for warnings as follow-up on intel/llvm#1583

Signed-off-by: Aleksander Fadeev aleksander.fadeev@intel.com

### CODE REVIEW

File: sycl/test/warnings/vec_convert_warnings.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+using namespace cl::sycl;
+int main() {
+  vec<long, 4> newVec;
+  queue myQueue;
+  buffer<vec<long, 4>, 1> resultBuf {&newVec, range<1> {1}};
+  myQueue.submit([&](handler& cgh) {
+    auto writeResult = resultBuf.get_access<access::mode::write>(cgh);
+    cgh.single_task<class kernel_name>([=]() {
+      writeResult[0] = (vec<int, 4>{1, 2, 3, 4}).template convert<long>();
```
Comment (bader) [inline sycl/test/warnings/vec_convert_warnings.cpp:14]: Please, extend warnings.cpp with this test case instead of adding another test.

Comment (fadeeval) [reply sycl/test/warnings/vec_convert_warnings.cpp:14]: Okey

File: sycl/test/warnings/warnings.cpp [hunk 2]
```diff
@@ -1,20 +1,17 @@
...
+int main() {
+  vec<long, 4> newVec;
+  queue myQueue;
+  buffer<vec<long, 4>, 1> resultBuf {&newVec, range<1> {1}};
+  myQueue.submit([&](handler& cgh) {
+    auto writeResult = resultBuf.get_access<access::mode::write>(cgh);
+    cgh.single_task<class kernel_name>([=]() {
+      writeResult[0] = (vec<int, 4>{1, 2, 3, 4}).template convert<long>();
     });
   });
```
Comment (s-kanaev) [inline sycl/test/warnings/warnings.cpp:15]: Are there any cases that were covered with old test and not covered by the new test?

Comment (fadeeval) [reply sycl/test/warnings/warnings.cpp:15]: I think no, there was just a simple program that launches the check of warnings.

Comment (fadeeval) [reply sycl/test/warnings/warnings.cpp:15]: But maybe we should ask the code owner.

**Comment by fadeeval** [comment]
@turinevgeny, approve, please, if no objections.

**Pull request merged**